gdal/frmts/hdf5/hdf5dataset.cpp

```diff
@@ -295,6 +308,9 @@ int HDF5Dataset::Identify( GDALOpenInfo * poOpenInfo )
 /************************************************************************/
 GDALDataset *HDF5Dataset::Open( GDALOpenInfo *poOpenInfo )
 {
+    vsi_l_offset filesz;
```

Comment:
please move filesz just before the use of the variable. Will save a warning about the variable not being used in the other case